topbeat/beater/config.go

```diff
@@ -1,17 +1,48 @@
 package beater
 
+import (
+	"errors"
+	"time"
+)
+
+type ConfigSettings struct {
+	Topbeat TopConfig `config:"topbeat"`
+}
+
 type TopConfig struct {
-	Period *int64
-	Procs  *[]string
-	Stats  struct {
-		System     *bool `config:"system"`
-		Proc       *bool `config:"process"`
-		Filesystem *bool `config:"filesystem"`
-		CpuPerCore *bool `config:"cpu_per_core"`
+	Period time.Duration     `config:"period"  validate:"min=1ms"`
+	Procs  []string          `config:"procs"`
+	Stats  StatsEnableConfig `config:"stats"`
+}
+
+type StatsEnableConfig struct {
+	System     bool `config:"system"`
+	Proc       bool `config:"process"`
+	Filesystem bool `config:"filesystem"`
+	CPUPerCore bool `config:"cpu_per_core"`
+}
+
+var (
+	defaultConfig = TopConfig{
+		Period: 10 * time.Second,
+		Procs:  []string{".*"}, //all processes
+		Stats: StatsEnableConfig{
+			System:     true,
+			Proc:       true,
+			Filesystem: true,
+			CPUPerCore: false,
+		},
 	}
+)
+
+func (c *TopConfig) Validate() error {
+	return nil
 }
 
-type ConfigSettings struct {
-	Input   *TopConfig `config:"input"`
-	Topbeat *TopConfig `config:"topbeat"`
+func (c *StatsEnableConfig) Validate() error {
```

Comment:
nested struct is not needed for `Validate` as `Topbeat` config also has validate method. Problem is line 29 defining the defaults.